game/src/backend/gamestate.rs

```diff
@@ -253,17 +253,15 @@ impl GameState {
                     self.player.resources_change.oxygen = -1;
                     self.player.resources_change.energy = -1;
                     self.player.last_damage = 0;
-                    self.events = None;
+                    self.events = Vec::new();
                     self.player.match_milestone = 1;
                 }
-                if ctx.time.ticks() % 5000 == 0 {
-                    if self.events.is_none() {
-                        self.events =
-                            Event::event_generator(self.screen_sender.as_ref().unwrap().clone())
-                    } else {
-                        self.events = Event::restore_event()
-                    }
-                }
+                Event::update_events(
+                    &ctx,
+                    &mut self.events,
+                    &mut self.player.resources_change,
+                    self.screen_sender.as_ref().unwrap().clone(),
```

Comment:
Können wir das nicht in den Gamestate schreiben, da wir hier drei Methoden haben, die self als reference nehmen? (TBF der is grade schon recht groß)